src/p11keygen.c

```diff
@@ -277,51 +279,56 @@ int main(int argc, char **argv) {
 
 	    case 'k':
 		if (strcasecmp(optarg, "aes") == 0) {
-		    keytype = aes;
-		    kb = 256;
+			if (keytype == unknown) { keytype = aes;}
```

Comment:
In all other instances we overwrite arguments that are repeated. Should'nt we stay consistent wrt to `k`?

Reply:
Looks like a leftover. Good catch.

Reply:
OK, got the issue. keytype must be set unconditionally under `'k'` option. Code adapted accordingly.